server/src/rooms/game/events/index.ts

```diff
@@ -184,7 +184,7 @@ export class RejectTradeRequest extends GameEventWithData {
     const toRole: Role = game.tradeSet[this.data.id].to.role;
     const fromRole: Role = game.tradeSet[this.data.id].from.role;
     
-    delete game.tradeSet[this.data.id];
+    game.removeTrade(this.data.id, 'reject-trade-request');
```

Comment:
This is a good start. Can we push more logic into removeTrade like the notifications etc?